File: app/src/mock/java/com/utynote/app/dependencies/NetModule.java
package com.utynote.app.dependencies;

import android.support.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;

import static com.utynote.utils.Preconditions.checkNotNull;

@Module
public class NetModule {

    private static Interceptor sInterceptor;

    @Provides
    @Singleton
    OkHttpClient provideOkHttpClient() {
        return new OkHttpClient.Builder()
                .addInterceptor(checkNotNull(sInterceptor, "NetModule is not properly configured"))
                .build();
    }

    public static void setInterceptor(@NonNull Interceptor interceptor) {
        sInterceptor = interceptor;
    }
}

File: app/src/mock/java/com/utynote/app/dependencies/NetModule.kt
package com.utynote.app.dependencies

import dagger.Module
import dagger.Provides
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import javax.inject.Singleton

@Module
class NetModule {

    @Provides
    @Singleton
    internal fun provideOkHttpClient(): OkHttpClient {
        return OkHttpClient.Builder()
                .addInterceptor(checkNotNull(sInterceptor))
                .build()
    }

    companion object {

        private var sInterceptor: Interceptor? = null

        fun setInterceptor(interceptor: Interceptor) {
            sInterceptor = interceptor
        }
    }
}
